src/pages/index.js

```diff
@@ -25,28 +36,135 @@ const cardData = {
 const userInfo = new UserInfo({
   nameSelector: ".profile__title",
   jobSelector: ".profile__description",
+  avatarSelector: ".profile__image",
 });
 
+api.getUserInfo().then((userData) => {
+  //console.log("Setting avatar to:", userData.avatar);
+  //console.log("User Data from API:", userData);
+  userInfo.setUserInfo({
+    name: userData.name,
+    job: userData.about,
+    avatar: userData.avatar,
+  });
+});
+
+/* ------------------------------------------------- */
+/*                     Form Validation
+/* ------------------------------------------------- */
+const formValidators = {};
+
+const enableValidation = (settings) => {
+  const formList = Array.from(document.querySelectorAll(settings.formSelector));
+  formList.forEach((formElement) => {
+    const validator = new FormValidator(settings, formElement);
+    const formName = formElement.getAttribute("name");
+
+    formValidators[formName] = validator;
+    validator.enableValidation();
+  });
+};
+
+enableValidation(settings);
+
 // /* ------------------------------------------------- */
-// /*                 Profile Edit Form
+// /*                 Profile Avatar
 /* ------------------------------------------------- */
+const handleAvatarFormSubmit = (formData) => {
+  const newAvatarUrl = formData.avatarUrl;
+
+  return api
+    .updateAvatar(newAvatarUrl)
+    .then((userData) => {
+      console.log("Avatar updated on server:", userData.avatar);
+      userInfo.setUserAvatar(userData.avatar);
+      profileEditAvatar.close();
```

Comment:
You don't need it because you close the modal inside `PopupWithForm`

**Please, correct it everywhere in the code**